block concurrent scrapes

ripienaar opened the pull request:
The design of this is not safe for multi threaded use, concurrent
access will publish concurrent pings and so we will get 2 sets
of replies - however only 1 polling boolean exist so this will
always confuse things - leading to logs about late replies etc

This will make concurrent polls just fail which is better than
give weird stats

ripienaar committed:
block concurrent scrapes

The design of this is not safe for multi threaded use, concurrent
access will publish concurrent pings and so we will get 2 sets
of replies - however only 1 polling boolean exist so this will
always confuse things - leading to logs about late replies etc

This will make concurrent polls just fail which is better than
give weird stats

ColinSullivan1 commented on surveyor/surveyor.go:
```diff
@@ -233,12 +233,37 @@ func (s *Surveyor) getScrapeHandler() http.Handler {
+
+		if sz.Polling() {
+			rw.WriteHeader(http.StatusServiceUnavailable)
+			rw.Write([]byte("Concurrent polls are not supported"))
```
Thanks!  This was needed.

ripienaar committed:
add tests for concurrency

Also fix some race conditions where polling would stay true when in
fact polling failed and we are not polling anymore

ripienaar committed:
remove debug logging
